### PR TITLE
Migrate to null safety

### LINKED ISSUE #53: Migrate to Null Safe dart.
Includes : 

- [ ] Upgrade packages
- [ ] Check all the variables/functions

Guide:- [https://dart.dev/null-safety/migration-guide](https://dart.dev/null-safety/migration-guide)

### PR DESCRIPTION
**Pull request by aman-singh7**
- [x] Upgrade dependencies who are opting into null safety.
- [x] Replace the dependencies which currently do not support null safety. 
- [x] migrate to null safety

fixes #53

### CODE REVIEW

**Comment by aman-singh7** [comment]
Dependencies that does not support null safety:-
- autocomplete_textfield

- battery_optimization

- flutter_swiper

- intent

- searchable_dropdown

**Review by praeclarumjj3 — COMMENTED** [review]
I will merge this one after the institute list search option is added on the register screen

**Comment by aman-singh7** [comment]
@praeclarumjj3 search box has been added in the institute dropdown search.

File: lib/screens/signup/signup.dart [hunk 1]
```diff
@@ -165,6 +164,7 @@ class _SignUpPageState extends State<SignUpPage> {
         borderRadius: const BorderRadius.all(Radius.circular(4.0)),
       ),
       child: DropdownSearch<String>(
+        showSearchBox: true,
```
Comment (praeclarumjj3) [inline lib/screens/signup/signup.dart:167]: Great, a single-line solution!

**Review by praeclarumjj3 — APPROVED** [review]

**Pull request merged**